lua/ttt2/libraries/gameloop.lua

```diff
@@ -0,0 +1,540 @@
+
+        -- starts the preparing phase timer that transitions from the preparing phase to the active round
+        timer.Create("prep2begin", timePrepPhase, 1, gameloop.Begin)
+
+        ---
+        -- @realm server
+        -- stylua: ignore
+        hook.Run("TTT2PrePrepareRound", timePrepPhase)
+
+        ---
+        -- @realm server
+        -- stylua: ignore
+        hook.Run("TTTPrepareRound")
+
+        LANG.Msg("round_begintime", { num = timePrepPhase })
+    end
+
+    function gameloop.Begin()
+        -- check for low-karma players that weren't kicked/banned on round end
+        KARMA.RoundBegin()
+
+        -- if not enough players are available, we go into a idle state where
+        -- the game periodically tries again to start a new round until enough
+        -- player were found
+        if gameloop.CheckForAbort() then
+            return
+        end
+
+        -- respawn players that died during the preparing phase
+        entspawn.SpawnPlayers(true)
+
+        -- remove their ragdolls as well
+        ents.TTT.RemoveRagdolls(true)
+
+        -- select role; this is where things really start so we can't abort anymore
+        -- this function also syncs the roles to all connected clients
+        roleselection.SelectRoles()
+
+        -- update the round state that is used to determine how the game behaves
+        SetRoundState(ROUND_ACTIVE)
+
+        -- triggers the round state output for every map setting entity
+        ents.TTT.TriggerRoundStateOutputs(ROUND_ACTIVE)
+
+        local timeRoundEnd = CurTime() + cvDurationRound:GetInt() * 60
+
+        if HasteMode() then
+            timeRoundEnd = CurTime() + cvTimeHasteStarting:GetInt() * 60
+
+            -- this is a "fake" time shown to innocents, showing the end time if no
+            -- one would have been killed, it has no gameplay effect
+            SetGlobalFloat("ttt_haste_end", timeRoundEnd)
+        end
+
+        gameloop.SetPhaseEnd(timeRoundEnd)
+
+        -- Start the win condition check timer
+        StartWinChecks() -- todo new function/s
+        StartNameChangeChecks() -- todo new function
+
+        -- todo: trigger the menu at round begin with role info
+        --timer.Simple(1.5, TellTraitorsAboutTraitors)
+        --timer.Simple(2.5, ShowRoundStartPopup)
+
+        ServerLog("Round proper has begun...\n")
+
+        ---
+        -- @realm server
+        -- stylua: ignore
+        hook.Run("TTT2PreBeginRound")
+
+        ---
+        -- @realm server
+        -- stylua: ignore
+        hook.Run("TTTBeginRound")
+    end
+
+    function gameloop.End(result)
+        KARMA.RoundEnd()
+
+        -- update the round state that is used to determine how the game behaves
+        SetRoundState(ROUND_POST)
+
+        -- triggers the round state output for every map setting entity
+        ents.TTT.TriggerRoundStateOutputs(ROUND_POST, result)
+
+        PrintResultMessage(result) -- todo what does this do?
```

Comment:
this just prints the winning team to the server console.

i would just leave it as is (for this PR at least)